test/integration_test/migration_test.go

```diff
@@ -203,3 +259,240 @@ func migrationLabelSelectorTest(t *testing.T) {
 		false,
 	)
 }
+
+/**** migration schedule tests *****/
+func migrationIntervalScheduleTest(t *testing.T) {
+	var err error
+	// Reset config in case of error
+	defer func() {
+		err = setRemoteConfig("")
+		require.NoError(t, err, "Error resetting remote config")
+	}()
+
+	ctxs, preMigrationCtx := triggerMigration(
+		t,
+		"mysql-migration-schedule-interval",
+		"mysql-1-pvc",
+		nil,
+		[]string{"mysql-migration-schedule-interval"},
+		true,
+	)
+
+	// bump time of the world by 5 minutes
+	mockNow := time.Now().Add(6 * time.Minute)
+	setMockTime(&mockNow)
+
+	validateAndDestroyMigration(t, ctxs, preMigrationCtx, true, false, true)
+}
+
+func migrationDailyScheduleTest(t *testing.T) {
+	migrationScheduleTest(t, v1alpha1.SchedulePolicyTypeDaily, "mysql-migration-schedule-daily", "", -1)
+}
+
+func migrationWeeklyScheduleTest(t *testing.T) {
+	migrationScheduleTest(t, v1alpha1.SchedulePolicyTypeWeekly, "mysql-migration-schedule-weekly", "Monday", -1)
+}
+
+func migrationMonthlyScheduleTest(t *testing.T) {
+	migrationScheduleTest(t, v1alpha1.SchedulePolicyTypeMonthly, "mysql-migration-schedule-monthly", "", 11)
+}
+
+func migrationScheduleInvalidTest(t *testing.T) {
+	migrationSchedules := []string{
+		"mysql-migration-schedule-daily-invalid",
+		"mysql-migration-schedule-weekly-invalid",
+		"mysql-migration-schedule-monthly-invalid",
+	}
+
+	ctxs, preMigrationCtx := triggerMigration(
+		t,
+		"mysql-migration-schedule-invalid",
+		"mysql-1-pvc",
+		nil,
+		migrationSchedules,
+		true,
+	)
+
+	namespace := preMigrationCtx.GetID()
+
+	// **** TEST ensure 0 migrations since the schedule is invalid. Also check events for invalid specs
+	for _, migrationScheduleName := range migrationSchedules {
+		migrationSchedule, err := k8s.Instance().GetMigrationSchedule(migrationScheduleName, namespace)
+		require.NoError(t, err, "failed to get migration schedule")
+		require.Empty(t, migrationSchedule.Status.Items, "expected 0 items in migration schedule status")
+
+		listOptions := meta_v1.ListOptions{
+			FieldSelector: fields.OneTermEqualSelector("involvedObject.name", migrationScheduleName).String(),
+			Watch:         false,
+		}
+
+		storkEvents, err := k8s.Instance().ListEvents(namespace, listOptions)
+		require.NoError(t, err, "failed to list stork events")
+
+		foundFailedEvent := false
+		for _, ev := range storkEvents.Items {
+			if ev.Reason == "Failed" && strings.Contains(ev.Message, "Invalid") {
```

Comment:
If you do want to validate the message in the event then check for more than just one word. Otherwise, this can lead to false positives from other error messages/